packages/sources/alongside/src/endpoint/collateral/index.ts

```diff
@@ -103,32 +109,118 @@ export class AlongsideCollateralTransport implements Transport<EndpointTypes> {
-      requestTradingBalance,
-    )
-    logger.debug('Requesting trading vault')
-    const vaultBalances = await this.requester.request<ProviderResponseBody>(
-      req.id,
-      requestTradingVault,
-    )
-    logger.debug('Getting asset weights')
-    const units = await collateral.getAssetWeights()
-    logger.debug('Calculating minimum collateral')
-    const result = collateral.calcMinCollateral(
-      tradingBalances.response.data.balances,
-      vaultBalances.response.data.balances,
-      units,
-    )
+
+    try {
+      logger.debug('Preparing request for trading balance')
+      const requestTradingBalance = this.prepareRequest('TRADING', context.adapterSettings)
+      logger.debug('Requesting trading balance')
+      tradingBalances = await this.requester.request<ProviderResponseBody>(
+        'trading_balance_request',
+        requestTradingBalance,
+      )
+    } catch (e) {
+      const errorMessage = 'Error occurred retrieving trading balance'
+      logger.error(errorMessage)
+      await this.responseCache.write(this.name, [
+        {
+          params: {},
+          response: {
+            statusCode: 502,
+            errorMessage,
+            timestamps: {
+              providerDataRequestedUnixMs,
+              providerDataReceivedUnixMs: Date.now(),
+              providerIndicatedTimeUnixMs: undefined,
+            },
+          },
+        },
+      ])
+      return
+    }
+    try {
+      logger.debug('Preparing request for vault balance')
+      const requestTradingVault = this.prepareRequest('VAULT', context.adapterSettings)
+      logger.debug('Requesting trading vault')
+      vaultBalances = await this.requester.request<ProviderResponseBody>(
+        'vault_balance_request',
+        requestTradingVault,
+      )
+    } catch (e) {
+      const errorMessage = 'Error occurred retrieving vault balance'
+      logger.error(errorMessage)
+      await this.responseCache.write(this.name, [
+        {
+          params: {},
+          response: {
+            statusCode: 502,
+            errorMessage,
+            timestamps: {
+              providerDataRequestedUnixMs,
+              providerDataReceivedUnixMs: Date.now(),
+              providerIndicatedTimeUnixMs: undefined,
+            },
+          },
+        },
+      ])
+      return
+    }
+
+    let units
+    try {
+      logger.debug('Getting asset weights')
+      units = await collateral.getAssetWeights()
+    } catch (e) {
+      const errorMessage = 'Error occurred retrieving asset weights'
+      logger.error(errorMessage)
+      await this.responseCache.write(this.name, [
+        {
+          params: {},
+          response: {
+            statusCode: 502,
+            errorMessage,
+            timestamps: {
+              providerDataRequestedUnixMs,
+              providerDataReceivedUnixMs: Date.now(),
+              providerIndicatedTimeUnixMs: undefined,
+            },
+          },
+        },
+      ])
+      return
```

Comment:
We're using this try catch pattern a lot and it makes it difficult to read the logic, let's abstract it away so we can do a single line await for the value